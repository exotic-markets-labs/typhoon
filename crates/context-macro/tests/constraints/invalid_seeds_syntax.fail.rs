use {
    bytemuck::{AnyBitPattern, NoUninit},
    pinocchio_pubkey::{declare_id, pinocchio::pubkey::Pubkey},
    typhoon_account_macro::*,
    typhoon_accounts::*,
    typhoon_context_macro::*,
    typhoon_program_id_macro::program_id,
};

program_id!("Fg6PaFpoGXkYsidMpWTK6W2BeZ7FEfcYkg476zPFsLnS");

#[derive(NoUninit, AnyBitPattern, AccountState, Copy, Clone)]
#[repr(C)]
pub struct Counter {
    pub count: u64,
}

#[context]
#[args(admin: Pubkey)]
pub struct InitContext {
    pub payer: Mut<Signer>,
    #[constraint(
        seeds = &[&args.admin],
        bump
    )]
    pub counter: Mut<Account<Counter>>,
    pub system: Program<System>,
}

pub fn main() {}
