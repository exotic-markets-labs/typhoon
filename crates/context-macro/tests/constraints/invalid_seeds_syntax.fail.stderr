error: expected square brackets
  --> tests/constraints/invalid_seeds_syntax.fail.rs:21:17
   |
21 |         seeds = &[&args.admin],
   |                 ^
